common/src/main/java/org/opensearch/ml/common/transport/upload/MLUploadInput.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.transport.upload;
+
+import lombok.Builder;
+import lombok.Data;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.ml.common.FunctionName;
+import org.opensearch.ml.common.model.MLModelConfig;
+import org.opensearch.ml.common.model.MLModelFormat;
+import org.opensearch.ml.common.model.TextEmbeddingModelConfig;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+/**
+ * ML input data: algirithm name, parameters and input data set.
+ */
+@Data
+public class MLUploadInput implements ToXContentObject, Writeable {
+
+    public static final String FUNCTION_NAME_FIELD = "function";
```

Comment:
What is this function field about? is this algorithm name?

Reply:
Yes, this part is a little bit messy. We call it algorithm name before. Now for custom model, it seems not so reasonable to call it "algorithm". So just use a more general name `function` here. Do you have any suggestion?